### PR TITLE
Order the values of the fields

### PR DESCRIPTION
**Pull request by llrs**
It is my first time working with R6 classes, and I'm a bit lost

Currently this fix works when the fields are well formatted (already in order and without double fields), but it doesn't work in the tests/testthat/D8 file. But I am unable to locate how should I change this. I also didn't include any further test...

If you point me towards a function or method I'll happily improve the PR.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Thanks! 

- is the other change needed as  well? I don't think so.
- please update the test cases, so they don't fail.
- please update the docs
- please add an entry to the NEWS file.

Thanks again!

**Comment by llrs** [comment]
With the new commits I address some of the latest points:

- [x] is the other change needed as well? I don't think so.
- [x]  please update the test cases, so they don't fail.
- [ ]  please update the docs: Where should that be in `desc_normalize`?
- [x]  please add an entry to the NEWS file.

**Comment by llrs** [comment]
@gaborcsardi I updated my fork and the documentation (I  hope I haven't messed up too many things). 
Is the pull request now in conditions to merge?

File: tests/testthat/test-authors.R [hunk 1]
```diff
@@ -40,6 +40,7 @@ test_that("we can add an author", {
 
 test_that("we can add an author with ORCID via comment", {
 
+
```
Comment (gaborcsardi) [inline tests/testthat/test-authors.R]: Can you please undo the whitespace changes? It is just commit noise. Also below. Thanks!

Comment (llrs) [reply tests/testthat/test-authors.R]: Done!

File: tests/testthat/test-authors.R [hunk 2]
```diff
@@ -172,7 +172,6 @@ test_that("we cannot add the same ORCID to more than one author", {
 
 })
 
-
```
Comment (gaborcsardi) [inline tests/testthat/test-authors.R]: Sorry,  there is still a whitespace change here.

Comment (llrs) [reply tests/testthat/test-authors.R]: Oh, I think I misunderstood the style

**Comment by gaborcsardi** [comment]
Thanks! Sorry for the nitpicking. Looks great now!

File: tests/testthat/test-authors.R (hunk 2 above)
Comment (gaborcsardi) [reply tests/testthat/test-authors.R]: I just prefer to avoid whitespace changes in general, even if they "fix" the style.

**Comment by gtm19** [comment]
This is still an issue (#66) for me. See reprex below, where `normalize()` does not sort dependencies:

``` r
desc <- desc::description$new("!new")

# Not in order:
desc$set_dep("lubridate", "Suggests")
desc$set_dep("magrittr", "Suggests")
desc$set_dep("here", "Suggests")
desc$set_dep("anytime", "Suggests")

desc$normalize()

ordered_desc <- desc::description$new("!new")

# Already in order:
ordered_desc$set_dep("anytime", "Suggests")
ordered_desc$set_dep("here", "Suggests")
ordered_desc$set_dep("lubridate", "Suggests")
ordered_desc$set_dep("magrittr", "Suggests")

desc
#> Package: {{ Package }}
#> Title: {{ Title }}
#> Version: 1.0.0
#> Authors@R (parsed):
#>     * Jo Doe <jodoe@dom.ain> [aut, cre]
#> Maintainer: {{ Maintainer }}
#> Description: {{ Description }}
#> License: {{ License }}
#> URL: {{ URL }}
#> BugReports: {{ BugReports }}
#> Suggests:
#>     lubridate,
#>     magrittr,
#>     here,
#>     anytime
#> Encoding: UTF-8
#> LazyData: true
ordered_desc
#> Package: {{ Package }}
#> Title: {{ Title }}
#> Version: 1.0.0
#> Authors@R (parsed):
#>     * Jo Doe <jodoe@dom.ain> [aut, cre]
#> Maintainer: {{ Maintainer }}
#> Description: {{ Description }}
#> License: {{ License }}
#> URL: {{ URL }}
#> BugReports: {{ BugReports }}
#> Suggests:
#>     anytime,
#>     here,
#>     lubridate,
#>     magrittr
#> Encoding: UTF-8
#> LazyData: true
```

<sup>Created on 2020-12-21 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
<details>
  <summary>Session Info</summary>

```r
sessionInfo()
#> R version 4.0.3 (2020-10-10)
#> Platform: x86_64-apple-darwin17.0 (64-bit)
#> Running under: macOS Catalina 10.15.7
#> 
#> Matrix products: default
#> BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
#> LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
#> 
#> locale:
#> [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] withr_2.3.0      rprojroot_2.0.2  digest_0.6.27    crayon_1.3.4    
#>  [5] assertthat_0.2.1 R6_2.5.0         magrittr_2.0.1   evaluate_0.14   
#>  [9] highr_0.8        rlang_0.4.9      stringi_1.5.3    testthat_3.0.0  
#> [13] rmarkdown_2.5    desc_1.2.0       tools_4.0.3      stringr_1.4.0   
#> [17] pkgload_1.1.0    xfun_0.19        yaml_2.2.1       compiler_4.0.3  
#> [21] htmltools_0.5.0  knitr_1.30
```

<sup>Created on 2020-12-21 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
</details>

**Comment by gaborcsardi** [comment]
I think this is only in the dev version.

**Comment by llrs** [comment]
Yes,  @gtm19, as Gábor said this PR wasn't merged before last release on CRAN. You'll need to install the github version or wait until next release.

**Pull request merged**